libvcx/src/api/vcx.rs

```diff
@@ -15,17 +15,117 @@ use crate::utils::error;
 use crate::utils::threadpool::spawn;
 use crate::utils::version_constants;
 
-/// Initializes VCX with config settings
+#[no_mangle]
+pub extern fn vcx_init_threadpool() -> u32 {
+    info!("vcx_init_threadpool >>>");
+    match init_threadpool() {
+        Ok(_) => error::SUCCESS.code_num,
+        Err(err) => error::UNKNOWN_ERROR.code_num
+    }
+}
+
+/// Creates an instance of agency client used to communicate with the agency. Must be called after
+/// wallet was created and opened.
 ///
 /// #Params
 /// command_handle: command handle to map callback to user context.
 ///
-/// config: The agent provision configuration. You can produce this by provisioning agent using function vcx_provision_agent
+/// agency_config: Config of the agency client
+/// {
+///    agency_url - URL of the agency 
+///    agency_did - DID of the agency
+///    agency_pwdid - pairwise DID of the agency created for the connection
+///    agency_vk - verkey  of the agency created for the connection
+///    agent_pwdid - pairwise DID of this client's agent in the agency
+///    agent_vk - verkey of this client's agent in the agency
+///    my_pwdid - pairwise DID of this client used to communicate with it's agent in the agency
+///    my_vk - verkey of this client used to communicate with it's agent in the agency
+/// }
 ///
-/// cb: Callback that provides error status of initialization
+/// cb: Callback that provides error status
+///
+/// #Returns
+/// Error code as a u32
+#[no_mangle]
+pub extern fn vcx_create_agency_client(command_handle: CommandHandle, config: *const c_char, cb: Option<extern fn(xcommand_handle: CommandHandle, err: u32)>) -> u32 {
```

Comment:
Let's "save" this good name `vcx_create_agency_client` for later :-) Let's use  `vcx_create_agency_client_for_main_wallet` - I know it's ugly, but the behaviour of this function is still linked to "singleton" nature of the library